test/CMakeLists.txt

```diff
@@ -272,7 +272,10 @@ function(Spglib_add_test test)
 	endforeach ()
 
 	add_test(NAME ${ARGS_TEST_NAME}
-			COMMAND ${CMAKE_CTEST_COMMAND} --build-and-test ${CMAKE_CURRENT_SOURCE_DIR}/${test}
+			COMMAND ${CMAKE_COMMAND} -E env --modify PATH=path_list_prepend:${CMAKE_INSTALL_PREFIX}/lib
```

Comment:
This should be in test property to make debugging more legible.

But for the usage in ctest --build-and-test I completely disagree with adding it. The example and package tests need to be able to test as standalone builds, like the user would have run it, and populating and ENV hides issues in thr CMakeLists.txt file that the user should know beforehand.

Note that testing-farm tests the case of standalone test project with system installed version. It does not cover Windows, but at the same time window does not have a packaging system we can upload to.

Reply:
This is the relevant part of the build log within visual studio 2020:
```
40/40 Testing: example-c_api
40/40 Test: example-c_api
Command: "C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/cmake.exe" "-E" "env"
 "--modify" "PATH=path_list_prepend:C:/Users/KLEIN_CL/Workspace/c/spglib/out/install/x64-debug/lib" "--"
 "C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/bin/ctest.exe"
 "--build-and-test" "C:/Users/KLEIN_CL/Workspace/c/spglib/test/example/c_api" "C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug/test/example/c_api"
 "--build-generator" "Ninja" "--build-options"
 "-D CMAKE_C_COMPILER=C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe"
 "-D CMAKE_CXX_COMPILER=C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe"
 "-D FETCHCONTENT_TRY_FIND_PACKAGE_MODE=ALWAYS"
 "-D Spglib_ROOT=C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug"
 "-D FETCHCONTENT_SOURCE_DIR_SPGLIB=C:/Users/KLEIN_CL/Workspace/c/spglib"
 "-D SPGLIB_WITH_Fortran=OFF" "-DCMAKE_PREFIX_PATH="
 "--test-command" "C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug/test/example/c_api/example" "&&"
 "C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug/test/example/c_api/example_full"
Directory: C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug/test/example
"example-c_api" start time: Feb 19 09:41 Mitteleuropäische Zeit
Output:
----------------------------------------------------------
Internal cmake changing into directory: C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug/test/example/c_api
======== CMake output     ======
The C compiler identification is MSVC 19.38.33135.0
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Professional/VC/Tools/MSVC/14.38.33130/bin/Hostx64/x64/cl.exe - skipped
Detecting C compile features
Detecting C compile features - done
Component omp of Spglib: NOT FOUND
Component fortran of Spglib: NOT FOUND
Found package: Spglib
Configuring done (5.3s)
Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_CXX_COMPILER
    FETCHCONTENT_SOURCE_DIR_SPGLIB
    FETCHCONTENT_TRY_FIND_PACKAGE_MODE
    SPGLIB_WITH_Fortran


Build files have been written to: C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug/test/example/c_api
======== End CMake output ======
Change Dir: 'C:/Users/KLEIN_CL/Workspace/c/spglib/out/build/x64-debug/test/example/c_api'

Run Clean Command: C:/RsPython/3.12/Scripts/ninja.exe clean
[1/1] Cleaning all built files...
Cleaning... 0 files.

Run Build Command(s): C:/RsPython/3.12/Scripts/ninja.exe
[1/4] Building C object CMakeFiles\example.dir\example.c.obj
[2/4] Building C object CMakeFiles\example_full.dir\example_full.c.obj
FAILED: CMakeFiles/example_full.dir/example_full.c.obj
C:\PROGRA~1\MIB055~1\2022\PROFES~1\VC\Tools\MSVC\1438~1.331\bin\Hostx64\x64\cl.exe  /nologo
-external:IC:\Users\KLEIN_CL\Workspace\c\spglib\out\build\x64-debug\src
-external:W0 /DWIN32 /D_WINDOWS /Ob0 /Od /RTC1 -std:c11 -MDd -Zi /showIncludes /FoCMakeFiles\example_full.dir\example_full.c.obj /FdCMakeFiles\example_full.dir\ /FS
-c C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(361): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(361): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(361): error C2133: 'rotation': unknown size
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(362): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(362): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(362): error C2133: 'translation': unknown size
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(392): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(392): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(392): error C2133: 'rotation': unknown size
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(393): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(393): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(393): error C2133: 'translation': unknown size
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(532): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(532): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(532): error C2133: 'grid_address': unknown size
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(533): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(533): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(533): error C2133: 'grid_mapping_table': unknown size
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(679): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(679): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(679): error C2133: 'pos': unknown size
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(680): error C2057: expected constant expression
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(680): error C2466: cannot allocate an array of constant size 0
C:\Users\KLEIN_CL\Workspace\c\spglib\test\example\c_api\example_full.c(680): error C2133: 'typ': unknown size
[3/4] Linking C executable example.exe
ninja: build stopped: subcommand failed.

<end of output>
Test time =   9.85 sec
----------------------------------------------------------
Test Failed.
"example-c_api" end time: Feb 19 09:41 Mitteleuropäische Zeit
"example-c_api" time elapsed: 00:00:09
----------------------------------------------------------
```